crates/ruff_python_formatter/src/trivia.rs

```diff
@@ -91,599 +90,11 @@ pub(crate) fn skip_trailing_trivia(offset: TextSize, code: &str) -> TextSize {
     offset
 }
 
-fn is_identifier_start(c: char) -> bool {
-    c.is_ascii_alphabetic() || c == '_' || is_non_ascii_identifier_start(c)
-}
-
-// Checks if the character c is a valid continuation character as described
-// in https://docs.python.org/3/reference/lexical_analysis.html#identifiers
-fn is_identifier_continuation(c: char) -> bool {
-    if c.is_ascii() {
-        matches!(c, 'a'..='z' | 'A'..='Z' | '_' | '0'..='9')
-    } else {
-        is_xid_continue(c)
-    }
-}
-
-fn is_non_ascii_identifier_start(c: char) -> bool {
-    is_xid_start(c)
-}
-
-#[derive(Clone, Debug, Eq, PartialEq, Hash)]
-pub(crate) struct Token {
-    pub(crate) kind: TokenKind,
-    pub(crate) range: TextRange,
-}
-
-impl Token {
-    pub(crate) const fn kind(&self) -> TokenKind {
-        self.kind
-    }
-
-    #[allow(unused)]
-    pub(crate) const fn range(&self) -> TextRange {
-        self.range
-    }
-
-    pub(crate) const fn start(&self) -> TextSize {
-        self.range.start()
-    }
-
-    pub(crate) const fn end(&self) -> TextSize {
-        self.range.end()
-    }
-}
-
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-pub(crate) enum TokenKind {
```

Comment:
I decided to move this to the shared `ruff_python_whitespace` crate, whose name is getting less and less appropriate.